internal/ast/ast.go

```diff
@@ -185,6 +185,12 @@ type (
 		Key   Expr
 		Value Expr
 	}
+
+	// UpdateExpr node represents a represents a update expression
+	UpdateExpr struct {
```

Comment:
I think that a better name for this would be `IncExpr`. The documentation should also be changed to "IncExpr node represents a variable increment expression"